pandas/tests/scalar/timestamp/test_timestamp.py

```diff
@@ -671,10 +671,10 @@ def test_constructor_invalid_frequency(self):
             Timestamp("2012-01-01", freq=[])
 
     @pytest.mark.parametrize("box", [datetime, Timestamp])
-    def test_depreciate_tz_and_tzinfo_in_datetime_input(self, box):
+    def test_raise_tz_and_tzinfo_in_datetime_input(self, box):
         # GH 23579
         kwargs = {"year": 2018, "month": 1, "day": 1, "tzinfo": utc}
-        with tm.assert_produces_warning(FutureWarning):
+        with pytest.raises(ValueError, match="Cannot pass a datetime or Timestamp"):
             Timestamp(box(**kwargs), tz="US/Pacific")
```

Comment:
can we also have a case with the tzinfo kwarg

Reply:
i think we need separate `pytest.raises` contexts (and the tzinfo kwarg is supposed to be a tzinfo object)